runtime/trampolines.m

```diff
@@ -201,7 +201,13 @@
 					}
 				}
 			} else {
-				xamarin_assertion_message ("Don't know how to marshal a return value of type '%s.%s'. Please file a bug with a test case at https://github.com/xamarin/xamarin-macios/issues/new\n", mono_class_get_namespace (r_klass), mono_class_get_name (r_klass)); 
+				char *fullname = xamarin_class_get_full_name (r_klass, exception_gchandle);
+				if (!strcmp (fullname, "System.IntPtr")) {
+					returnValue =  *(void **) mono_object_unbox (retval);
+				} else {
+					xamarin_assertion_message ("Don't know how to marshal a return value of type '%s.%s'. Please file a bug with a test case at https://github.com/xamarin/xamarin-macios/issues/new\n", mono_class_get_namespace (r_klass), mono_class_get_name (r_klass));
```

Comment:
Since you have the `fullname`, why not change the assertion message to:
`xamarin_assertion_message ("Don't know how to marshal a return value of type '%s', Please Please file a bug with a test case at https://github.com/xamarin/xamarin-macios/issues/new\n", full name);`

On a side note, I wonder if this won't be immediately stale with the .NET 6 transition to `NativeHandle` instead of `IntPtr`.

Reply:
Both good points @stephen-hawley I'll address them shortly. Thanks!